python-checks/src/main/java/org/sonar/python/checks/cdk/PublicNetworkAccessToCloudResourcesCheck.java

```diff
@@ -51,10 +54,58 @@ protected void registerFqnConsumer() {
     checkFqn("aws_cdk.aws_rds.DatabaseInstance", PublicNetworkAccessToCloudResourcesCheck::checkDatabaseInstance);
 
     checkFqn("aws_cdk.aws_rds.CfnDBInstance", (subscriptionContext, callExpression) ->
-      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(
-        argument -> argument.addIssueIf(isTrue(), MESSAGE)
-      )
-    );
+      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(argument -> {
+        if (argument.hasExpression(isTrue()) && !isSafeNetwork(subscriptionContext, callExpression)) {
+          argument.addIssue(MESSAGE);
+        }
+      }));
```

Comment:
```suggestion
      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME)
        .filter(argument -> argument.hasExpression(isTrue()) && !isSafeNetwork(subscriptionContext, callExpression))
        .ifPresent(argument -> argument.addIssue(MESSAGE)));
```
Instead of having a condition inside the ifPresent block you can have it as a filter.